contracts/on-chain-claim/src/contract.rs

```diff
@@ -0,0 +1,141 @@
+#![no_std]
+#![allow(unused_attributes)]
+
+pub use address_info::AddressInfo;
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+pub mod address_info;
+pub mod config;
+
+use crate::config::{MAX_REPAIR_GAP, SFT_AMOUNT};
+use multiversx_sc_modules::only_admin;
+
+#[multiversx_sc::contract]
+pub trait OnChainClaimContract: config::ConfigModule + only_admin::OnlyAdminModule {
+    #[init]
+    fn init(&self, repair_streak_token_id: TokenIdentifier) {
+        require!(
+            repair_streak_token_id.is_valid_esdt_identifier(),
+            "Invalid token ID"
+        );
+        self.repair_streak_token_identifier()
+            .set(repair_streak_token_id);
+        let caller = self.blockchain().get_caller();
+        self.add_admin(caller);
+    }
+
+    #[upgrade]
+    fn upgrade(&self) {}
+
+    #[endpoint(claim)]
+    fn claim(&self) {
+        let caller = self.blockchain().get_caller();
+        require!(
+            !self.blockchain().is_smart_contract(&caller),
+            "Only user accounts can perform claim"
+        );
+        self.require_same_shard(&caller);
+
+        let current_epoch = self.blockchain().get_block_epoch();
+
+        let address_info_mapper = self.address_info(&caller);
+        if address_info_mapper.is_empty() {
+            let address_info = AddressInfo::new(1, current_epoch, 1, 1);
+            self.address_info(&caller).set(address_info);
+            return;
+        }
+
+        address_info_mapper.update(|address_info| {
+            require!(
+                address_info.last_epoch_claimed < current_epoch,
+                "epoch already claimed"
+            );
+
+            if address_info.last_epoch_claimed + 1 == current_epoch {
+                address_info.current_streak += 1;
+            } else {
+                address_info.current_streak = 1;
+            }
+
+            address_info.total_epochs_claimed += 1;
+            address_info.last_epoch_claimed = current_epoch;
+
+            if address_info.best_streak < address_info.current_streak {
+                address_info.best_streak = address_info.current_streak;
+            }
+        });
+    }
+
+    #[payable("*")]
+    #[endpoint(claimAndRepair)]
+    fn claim_and_repair(&self) {
+        let caller = self.blockchain().get_caller();
+        require!(
+            !self.blockchain().is_smart_contract(&caller),
+            "Only user accounts can perform claim and repair"
+        );
+        self.require_same_shard(&caller);
+
+        let payment = self.call_value().single_esdt();
+        let repair_streak_token_identifier = self.repair_streak_token_identifier().get();
+        require!(
+            payment.token_identifier == repair_streak_token_identifier,
+            "Bad payment token"
+        );
+        require!(payment.amount == SFT_AMOUNT, "Bad payment amount");
```

Comment:
Might as well have this as configurable value rather than a const.

Reply:
SFT_AMOUNT will always be 1, I don't believe we need this configurable